lib/galaxy/webapps/galaxy/config_schema.yml

```
@@ -2323,6 +2323,22 @@ mapping:
           stable API. (The module state definitions may change and workflows built using
           these modules may not function in the future.)
 
+      enable_beta_edam_toolbox:
+        type: bool
+        default: false
+        required: false
+        desc: |
+          Enable beta EDAM organised toolbox which ignores admin managed
+          sections in lieu of EDAM topics only.
+
+      beta_edam_toolbox_ontology_path:
+        type: str
+        default: EDAM_1.23.tsv
+        path_resolves_to: config_dir
+        required: false
+        desc: |
+          Sets the path to EDAM ontology file.
+
```

```suggestion
      beta_edam_toolbox_ontology_path:
        type: str
        default: EDAM_1.23.tsv
        path_resolves_to: data_dir
        required: false
        desc: |
          Sets the path to EDAM ontology file.

      beta_edam_ontology_url:
        type: str
        default: <url_to_wherever we can fetch the ontology>
        required: false
        desc: |
          Sets the url to the EDAM ontology file.
```
And then we can fetch that on startup if it's not present.

---

Ah, that's a nice suggestion, sure. sounds good.

---

Just to confirm because the diff looks odd: swap path for url, right? remove path completely?

---

I think we want to keep both. If the path doesn't exist we download and store the file at the path.

---

Why do we want the URL to be configurable? Can't we just download the latest version from http://edamontology.org/EDAM.tsv ?

---

In case EDAM releases a new ontology and people want to update before we update galaxy?

---

That URL is supposed to always point to the latest EDAM release.

---

Oh, sorry, misread the URL. Then I guess we need to worry about caching a bit, or? People will have the edam from whenever they first try this setting, and it won't be updated?

---

I thought there might be other places / custom ontologies people may want to load. If that's not true we can just make the URL a constant, sure.

---

> Oh, sorry, misread the URL. Then I guess we need to worry about caching a bit, or? People will have the edam from whenever they first try this setting, and it won't be updated?

Sure, we have a similar issue with the conda binary.

> I thought there might be other places / custom ontologies people may want to load. If that's not true we can just make the URL a constant, sure.

Given we are using the EDAM annotations in tools, I don't think we need to make that configurable at this stage.